### ISSUE TITLE
install on yaourt not working

### ISSUE THREAD

**Issue by wontruefree** [amberframework/amber#441]
### Description

when using either the commandline or the add/remove software I get and error about IO
```crystal
in lib/cli/src/lib/ios/pipe.cr:10: IO is not a module, it's a class

    include IO
```

### Steps to Reproduce

1. run `yaourt -S amber`
**Expected behavior:** 
install should finish without errors

**Actual behavior:**
install breaks and cannot finish

**Reproduces how often:** 
happens every time for me.

**Comment by faustinoaq** [comment]
Hi @wontruefree I can reproduce it, the main issue is that latest amber `v0.3.6` isn't compatible with crystal `v0.24.0` yet.

> **Note:** crystal `v0.24.0` was a pre-release the latest stable version by the time I'm writing this comment is still `v0.23.1`. This package upgrade on archlinux was totally unexpected.

Try downgrading crystal to `v0.23.1` downloading https://archive.archlinux.org/repos/2017/11/01/community/os/x86_64/crystal-0.23.1-2-x86_64.pkg.tar.xz

And then:

```
➜  ~ sudo pacman -U crystal-0.23.1-2-x86_64.pkg.tar.xz
loading packages...
warning: crystal-0.23.1-2 is up to date -- reinstalling
resolving dependencies...
looking for conflicting packages...

Packages (1) crystal-0.23.1-2

Total Installed Size:  65.15 MiB
Net Upgrade Size:       0.00 MiB

:: Proceed with installation? [Y/n] Y
(1/1) checking keys in keyring            [#############################################] 100%
(1/1) checking package integrity          [#############################################] 100%
(1/1) loading package files               [#############################################] 100%
(1/1) checking for file conflicts         [#############################################] 100%
(1/1) checking available disk space       [#############################################] 100%
:: Processing package changes...
(1/1) reinstalling crystal                [#############################################] 100%
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
➜  ~ crystal -v
Crystal 0.23.1 (2017-09-10) LLVM 5.0.0
```

Finally:

```
➜  ~ yaourt -S amber                                            

==> Downloading amber PKGBUILD from AUR...
x .SRCINFO
x .gitignore
x LICENSE
x PKGBUILD
x README.md
faustinoaq commented on 2017-09-21 15:26                         
Currently Amber releases are compiled from source.
Future releases will download just a binary (may be static linked).

amber 0.3.6-1  (2017-11-18 03:27)
( Unsupported package: Potentially dangerous ! )
==> Edit PKGBUILD ? [Y/n] ("A" to abort)
==> ------------------------------------
==> n

==> amber dependencies:
 - crystal (already installed)
 - shards (already installed)
 - sqlite (already installed)


==> Continue building amber ? [Y/n]
==> -------------------------------
==> Y

==> Building and installing package
==> Making package: amber 0.3.6-1 (Wed Dec 13 02:05:34 EST 2017)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading v0.3.6.tar.gz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   128    0   128    0     0    128      0 --:--:-- --:--:-- --:--:--   232
100  149k    0  149k    0     0   149k      0 --:--:--  0:00:01 --:--:--  403k
==> Validating source files with sha256sums...
    v0.3.6.tar.gz ... Passed
==> Extracting sources...
  -> Extracting v0.3.6.tar.gz with bsdtar
==> Starting build()...
Fetching https://github.com/luislavena/radix.git
Fetching https://github.com/jeromegn/kilt.git
Fetching https://github.com/jeromegn/slang.git
Fetching https://github.com/stefanwille/crystal-redis.git
Fetching https://github.com/mosop/cli.git
Fetching https://github.com/mosop/optarg.git
Fetching https://github.com/mosop/callback.git
Fetching https://github.com/mosop/string_inflection.git
Fetching https://github.com/mosop/teeplate.git
Fetching https://github.com/samueleaton/sentry.git
Fetching https://github.com/juanedi/micrate.git
Fetching https://github.com/crystal-lang/crystal-db.git
Fetching https://github.com/jwaldrip/shell-table.cr.git
Fetching https://github.com/askn/spinner.git
Fetching https://github.com/will/crystal-pg.git
Fetching https://github.com/crystal-lang/crystal-mysql.git
Fetching https://github.com/crystal-lang/crystal-sqlite3.git
Installing radix (0.3.8)
Installing kilt (0.4.0)
Installing slang (1.7.0)
Installing redis (1.9.0)
Installing cli (0.6.10)
Installing optarg (0.5.8)
Installing callback (0.6.3)
Installing string_inflection (0.2.1)
Installing teeplate (0.4.5)
Installing sentry (0.1.1)
Installing micrate (0.3.0)
Installing db (0.4.3)
Installing shell-table (0.9.2)
Installing spinner (0.1.1)
Installing pg (0.13.4)
Installing mysql (0.3.3)
Installing sqlite3 (0.8.3)
Building amber in /tmp/yaourt-tmp-main/aur-amber/src/amber-0.3.6
==> Entering fakeroot environment...                  
==> Starting package()...
==> Tidying install...
  -> Removing libtool files...
  -> Purging unwanted files...
  -> Removing static library files...
  -> Stripping unneeded symbols from binaries and libraries...
  -> Compressing man and info pages...
==> Checking for packaging issue...
==> Creating package "amber"...
  -> Generating .PKGINFO file...
  -> Generating .BUILDINFO file...
  -> Generating .MTREE file...
  -> Compressing package...
==> Leaving fakeroot environment.
==> Finished making: amber 0.3.6-1 (Wed Dec 13 02:06:03 EST 2017)
==> Cleaning up...
==> Exporting amber to /var/cache/pacman/pkg/ directory
'/tmp/yaourt-tmp-main/PKGDEST.qDw/amber-0.3.6-1-x86_64.pkg.tar.xz' -> 
'/var/cache/pacman/pkg/amber-0.3.6-1-x86_64.pkg.tar.xz'

==> Continue installing amber ? [Y/n]
==> [v]iew package contents [c]heck package with namcap
==> ---------------------------------------------------
==> y

loading packages...
resolving dependencies...
looking for conflicting packages...

Packages (1) amber-0.3.6-1

Total Installed Size:  5.48 MiB

:: Proceed with installation? [Y/n] y
(1/1) checking keys in keyring      [#############################################] 100%
(1/1) checking package integrity    [#############################################] 100%
(1/1) loading package files         [#############################################] 100%
(1/1) checking for file conflicts   [#############################################] 100%
(1/1) checking available disk space [#############################################] 100%
:: Processing package changes...
(1/1) installing amber              [#############################################] 100%
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
➜  ~ amber -v
Amber CLI (amberframework.org) - v0.3.6
```

I think next amber version will be compatible with crystal `v0.24.1`, see: 

- https://github.com/crystal-lang/crystal/pull/5358
- https://github.com/amberframework/amber/pull/358

**Comment by wontruefree** [comment]
it looks like yaourt is installing the 0.24.0.
I am looking into downgrading but it is interesting that yaourt is installing a prerelease.

**Comment by wontruefree** [comment]
I have started looking into this and it seems like a fair amount of downstream libs are effected.

**Comment by wontruefree** [comment]
is there a reasons why this is closed?

**Comment by faustinoaq** [comment]
> I am looking into downgrading but it is interesting that yaourt is installing a prerelease

Yaourt not but pacman, Crystal `v0.24.0` was mistakenly published on arch community.

This will be fixed on #453

**Comment by elorest** [comment]
@wontruefree 

I believe this issue was closed because @faustinoaq's solution works for @eliasjpr and I. Does it not work for you?
